analyzers/tests/SonarAnalyzer.UnitTest/Helpers/TypeHelperTest.cs

```diff
@@ -89,5 +89,412 @@ public void Type_GetSymbolType_Alias()
             symbol.ToString().Should().Be("PropertyBag");
             type.ToString().Should().Be("System.Collections.Generic.Dictionary<string, object>");
         }
+
+        [DataTestMethod]
+        [DataRow("int")]
+        [DataRow("System.Int32")]
+        [DataRow("int?")]
+        [DataRow("System.Nullable<int>")]
+        [DataRow("CustomStruct")]
+        [DataRow("CustomRefStruct")]
+        [DataRow("RecordStruct")]
+        public void IsStruct_Simple(string type)
+        {
+            var fieldSymbol = FirstFieldSymbolFromCode($$"""
+                struct CustomStruct { }
+                record struct RecordStruct { }
+                ref struct CustomRefStruct { }
+
+                ref struct Test
+                {
+                    {{type}} field;
+                }
+                """);
+            fieldSymbol.Type.IsStruct().Should().BeTrue();
+        }
+
+        [DataTestMethod]
+        [DataRow("object")]
+        [DataRow("System.IComparable")]
+        public void IsStruct_False_Simple(string type)
```

Comment:
To have the discriminator last
```suggestion
        public void IsStruct_Simple_False(string type)
```